vivisect/tests/testvivisect.py

```diff
@@ -1057,27 +1017,37 @@ def test_chgrp_exports(self):
         vw = self.chgrp_vw
         exports = [
             # .rodata
-            (0x08050650, 4, v_const.LOC_POINTER, 'chgrp._fp_hw'),
-            (0x08050654, 4, v_const.LOC_POINTER, 'chgrp._IO_stdin_used'),
+            (0x08050650, 4, v_const.LOC_NUMBER, 'chgrp._fp_hw'),  # 08050650  uint32_t _fp_hw = 0x3
+            (0x08050654, 4, v_const.LOC_NUMBER, 'chgrp._IO_stdin_used'),  # 08050654  uint32_t _IO_stdin_used = 0x20001
 
             # long options is really an array of pointers, but that isn't obvi
             # (0x08050748, 4, v_const.LOC_POINTER, 'chgrp.long_options'),
             #(0x08050850, 0x27, v_const.LOC_STRING, None),
 
-            (0x8054170, 4, v_const.LOC_POINTER, 'chgrp.program_invocation_short_name@@GLIBC_2.0'),
-            (0x8054178, 4, v_const.LOC_POINTER, 'chgrp.stderr@@GLIBC_2.0'),
-            (0x8054180, 4, v_const.LOC_POINTER, 'chgrp.program_invocation_name@@GLIBC_2.0'),
-            (0x8054190, 4, v_const.LOC_POINTER, 'chgrp.optind@@GLIBC_2.0'),
-            (0x8054194, 4, v_const.LOC_POINTER, 'chgrp.stdout@@GLIBC_2.0'),
-            (0x8054198, 4, v_const.LOC_POINTER, 'chgrp.optarg@@GLIBC_2.0'),
+            # these are zero-initialized global pointers, like:
+            #
+            #     uint32_t* const __progname = 0x0
+            #
+            # however, we can't differentiate this from a zero-initialized variable,
+            # especially without data flow analysis, 
+            # so we assume (incorrectly) that these are numbers rather than pointers.
+            # (0x8054170, 4, v_const.LOC_POINTER, 'chgrp.program_invocation_short_name@@GLIBC_2.0'),
+            # (0x8054178, 4, v_const.LOC_POINTER, 'chgrp.stderr@@GLIBC_2.0'),
+            # (0x8054180, 4, v_const.LOC_POINTER, 'chgrp.program_invocation_name@@GLIBC_2.0'),
+            # (0x8054190, 4, v_const.LOC_POINTER, 'chgrp.optind@@GLIBC_2.0'),
+            # (0x8054194, 4, v_const.LOC_POINTER, 'chgrp.stdout@@GLIBC_2.0'),
+            # (0x8054198, 4, v_const.LOC_POINTER, 'chgrp.optarg@@GLIBC_2.0'),
+            #
+            # this, on the other hand, is a zero-initialized global variable,
+            # which we can correctly identify.
             (0x805419c, 1, v_const.LOC_NUMBER, 'chgrp.completed.7282'),
```

Comment:
As commented, these changes do represent a regression. With data flow analysis, we can tell that these are zero-initialized global pointers that are updated at runtime. It's unfortunate that we can't tell at ELF parse time, but on balance, I believe we should interpret this data as numbers, until its clear a NULL pointer is more correct.